src/Datadog.Trace.ClrProfiler.Native/cor_profiler.h

```diff
@@ -16,16 +16,15 @@ namespace trace {
 
 const DWORD kEventMask =
     COR_PRF_MONITOR_JIT_COMPILATION |
-    COR_PRF_DISABLE_TRANSPARENCY_CHECKS_UNDER_FULL_TRUST | /* helps the case
-                                                                where this
-                                                                profiler is
-                                                                used on Full
-                                                                CLR */
+    COR_PRF_DISABLE_TRANSPARENCY_CHECKS_UNDER_FULL_TRUST |
     COR_PRF_DISABLE_INLINING |
+#ifndef  _WIN32
+    // workaround for JIT compiler bug in linux/amd64 and linux/arm64
+    // https://github.com/dotnet/coreclr/issues/24676
+    // https://github.com/dotnet/coreclr/issues/12468
+    COR_PRF_DISABLE_OPTIMIZATIONS |
```

Comment:
This is the core of the changes in this PR. Disable fast tail calls on Linux due to a JIT compiler bug.

Reply:
In what scenarios is _WIN32 not set? Is this all Linux builds or only a subset of Linux builds?

Reply:
`_WIN32` is a [predefined macro](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros) in MSVC++.

> _WIN32 Defined as 1 when the compilation target is 32-bit ARM, 64-bit ARM, x86, or x64. Otherwise, undefined.